src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -39,39 +39,60 @@ public ReadMetadata( final Set<Integer> crossContigIgnoreSet,
         this.crossContigIgnoreSet = crossContigIgnoreSet;
         contigNameToID = buildContigNameToIDMap(header);
 
-        final int nReadGroups = header.getReadGroups().size();
+        final List<SAMReadGroupRecord> readGroups = header.getReadGroups();
+        final int mapCapacity = SVUtils.hashMapCapacity(header.getReadGroups().size());
+        final Map<String, String> readGroupToLibraryMap = new HashMap<>(mapCapacity);
+        for ( final SAMReadGroupRecord groupRecord : readGroups ) {
+            readGroupToLibraryMap.put(groupRecord.getId(), groupRecord.getLibrary());
+        }
         final List<PartitionStatistics> perPartitionStatistics =
                 reads.mapPartitions(readItr ->
-                        Collections.singletonList(new PartitionStatistics(readItr, nReadGroups)).iterator())
+                        Collections.singletonList(new PartitionStatistics(readItr, readGroupToLibraryMap)).iterator())
                      .collect();
-        nPartitions = perPartitionStatistics.size();
         nReads = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNReads).sum();
         maxReadsInPartition = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNReads).max().orElse(0L);
         final long nReadBases = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNBases).sum();
         final long nRefBases = header.getSequenceDictionary().getSequences()
                 .stream().mapToLong(SAMSequenceRecord::getSequenceLength).sum();
-        coverage = (int) ((nReadBases + nRefBases - 1) / nRefBases);
-        readGroupToFragmentStatistics = new HashMap<>(SVUtils.hashMapCapacity(header.getReadGroups().size()));
+        coverage = (int) ((nReadBases + nRefBases/2) / nRefBases);
```

Comment:
Why this change to the coverage calculation? Can you put in a comment explaining the calculation?